08-rails/requests-and-responses.md

```diff
@@ -33,46 +93,95 @@ In the table below, you can see the starter mapping between the HTTP Verb and th
 
-## CRUD
-<b>C</b>reate. <b>R</b>ead. <b>U</b>pdate. <b>D</b>elete. This acronym embodies a huge majority of what we do with our web applications both simple and complex. We begin architecting our applications routes and interfaces by considering what kind of actions users will perform (CRUD) and what effects those actions will have on resources within the application (idempotency).
+Again, we can see there's a bunch of other HTTP stuff in there that we're ignoring for right now.
 
-## Summary
+### Status Codes
+
+Status codes are an important part of the HTTP protocol - they let the client know how things went. Every response must have a status code attached.
+
+Each status code is a 3-digit number paired with an English word or phrase. The first digit of the number indicates what sort of status it is, and the last two tell the details. It's sort of like how the first two digits of a room number are the floor that room is on.
+
+Here are some common status codes:
 
+Code    | Text             | Meaning
+---     | ---              | ---
+**2XX** | **Success**      |
+200     | OK               | Standard success response, typically has an HTML payload (common for `GET` requests)
+201     | Created          | We made a new resource for you (common for `POST` requests)
+204     | No Content       | We did what you asked and it went fine, but there's no HTML to give you (common for `DELETE` requests)
+**3XX** | **Redirection**  |
+302     | Found            | Go look at this other URL
+**4XX** | **Client Error** |
+400     | Bad Request      | The data you sent doesn't make sense
+401     | Unauthorized     | You need to log in to do that
+403     | Forbidden        | The person you're logged in as doesn't have permission to do that
+404     | Not Found        | The resource you requested doesn't exist
+**5XX** | **Server Error** |
+500     | Internal Server Error | There's a bug in this Rails project. As engineers, we see this one a lot!
+
+So far, we've let Rails handle sending the response for us, and by default Rails will send back a `200 OK` status code along with the HTML from the view. If we want to send something different, we have to say so explicitly.
+
+**Question:** Which part of Rails should be responsible for setting the status code of a response?
+
+There are a few ways to send back a status code from the controller, but here is one:
+
+```ruby
+head :not_found
+```
+
+When added to a controller action, the `head` method indicates that only HTTP headers should be sent back (i.e. no payload). The only argument, the symbol `:not_found`, is the text-version of the status code to use (404).
+
+Typically you would do this inside of a conditional:
+
+```ruby
+def show
+  book_id = params[:id]
+  @book = BOOKS[book_id]
+  if @book.nil?
+    head :not_found
```

Comment:
Does `head` allow the rest of the page to render?  If not should you use `render` instead?